Automattic/vip-go-mu-plugins
# Update/search default meta whitelist

## Description


## Changelog Description

### Plugin Updated: Search

In order to make future migration from Jetpack search easier, we implemented support for a `jetpack_sync_post_meta_whitelist` filter.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1. Create post with some meta - `lando wp post create --post_title="TEst title" --post_content="Content" --post_status="publish" --meta_input='{"foo": 1, "bar": 2}'`
1. Check meta - `lando wp vip-search documents get post <id> --format=json | jq ".[].meta"` - no meta should be present.
1. Add a vip-search allow list filter somewhere to your site `\add_filter('vip_search_post_meta_allow_list', function( $keys ) { return array_merge($keys, [ 'foo' ]); }, 0);`
1. Reindex - `lando wp vip-search index`
1. Check meta - should return **only** the one from your filter (`foo` in exable above)
1. Add jetpack filter - `\add_filter('jetpack_sync_post_meta_whitelist', function( $keys ) { return array_merge($keys, [ 'bar' ]); }, 0);`
1. Reindex
1. Check meta  - should return also the one form jetpack filter (both `foo` and `bar` from example above).

Note: Make sure your `vip_search_post_meta_allow_list` adds new values and does not ignore the defaults or remove it all together.
